gitfile/scmprovider.go

```diff
@@ -1,12 +1,14 @@
 package gitfile
 
-import "fmt"
+import (
+	"fmt"
+)
 
 // ScmProvider defines the interface that in order to determine if URL belongs to SCM provider
 type ScmProvider interface {
 	// Detect will detect whether the string matches a known SCM file pattern
 	// and transform it to valid https url.
-	Detect(repoUrl, filepath, ref string) (bool, string, error)
+	Detect(repoUrl, filepath, ref string, v ...interface{}) (bool, string, error)
```

Comment:
What's `ref` and `v`? As a user of a library I would be lost what to supply here. Is the `repoUrl` or the `filepath` the "string" that is supposed to match?

Reply:
ideally, this method should not be called from outside. but i'll add the missing specs.